test/Microsoft.AspNet.Mvc.FunctionalTests/TagHelpersTest.cs

```diff
@@ -148,5 +148,110 @@ public class TagHelpersTests
             var responseContent = await response.Content.ReadAsStringAsync();
             Assert.Equal(expectedContent, responseContent);
         }
+
+        [Fact]
+        public async Task CacheTagHelper_CanCachePortionsOfViewsPartialViewsAndViewComponents()
+        {
+            // Arrange
+            var server = TestServer.Create(_provider, _app);
+            var client = server.CreateClient();
+            client.BaseAddress = new Uri("http://localhost");
+            client.DefaultRequestHeaders.Add("Locale", "N");
+
+            // Act - 1
+            // Verify that content gets cached based on vary-by-params
```

Comment:
how does a single request verify caching?  most obvious caching indicators I see are down at lines https://github.com/aspnet/Mvc/pull/1778/files#diff-c00e94ab46fad24feae5b82d430aa45fR197 and https://github.com/aspnet/Mvc/pull/1778/files#diff-c00e94ab46fad24feae5b82d430aa45fR227